lib/bolt/shell/bash.rb

```diff
@@ -345,6 +345,23 @@ def execute(command, sudoable: false, **options)
         # together multiple commands into a single sh invocation
         commands = [inject_interpreter(options[:interpreter], command)]
 
+        # Only do this for the local transport, not if bundled-ruby is nil
+        if target.transport_config['bundled-ruby'] == false
+          ruby_env_vars = RUBY_ENV_VARS.each_with_object({}) do |e, acc|
+            acc[e] = ENV["BOLT_ORIG_#{e}"] if ENV["BOLT_ORIG_#{e}"]
```

Comment:
Wont this "pollute" the ENV for the rest of the bolt process? I was thinking that the `Connection` would just set these when we shell out with Open3.

Reply:
oh, i see, we are not actually setting these, just reading. Nevermind.

Reply:
Er, this is merging the values of `BOLT_ORIG` with the user-specified environment variables to get prepended to the command itself, not setting it *in* the Bolt environment. Or am I misunderstanding what you mean?

I initially had it in `Open3`, but was testing with Bundler using the `GEM_PATH` env var which [it explicitly unsets unless otherwise configured](https://github.com/rubygems/rubygems/blob/39771027ac3727a3234118a9795c733323d2188d/bundler/lib/bundler.rb#L671), but I didn't figure that out until I had moved it to be merged with the user-specified environment variables (thinking open3 might be the issue). I didn't think it mattered when I put this up, but I realized this *will* set the environment variables when using `run-as` on the local transport which I don't think we want, so I'll move it back to be in the connection later this morning (it honestly makes more sense there anyway).

Reply:
My first comment was a mistake. I thought we were *setting* ENV, but we are just reading from it. Sorry i was kind of half looking at this PR yesterday. I really want to get this in a build with ACE server but I wont have time to get that done this week. This general thing should totally work there though :)